pkg/testing/fixture_install.go

```diff
@@ -57,13 +57,25 @@ type InstallOpts struct {
 	Insecure       bool   // --insecure
 	NonInteractive bool   // --non-interactive
 	ProxyURL       string // --proxy-url
-	Unprivileged   bool   // --unprivileged
 	DelayEnroll    bool   // --delay-enroll
 
+	// Unprivileged by default installs the Elastic Agent as `--unprivileged` unless
+	// the platform being tested doesn't currently support it, or it's explicitly set
+	// to false.
+	Unprivileged *bool // --unprivileged
+
 	EnrollOpts
 }
 
-func (i InstallOpts) toCmdArgs() []string {
+func (i InstallOpts) IsUnprivileged(operatingSystem string) bool {
+	if i.Unprivileged == nil {
+		// not explicitly set, default to true on Linux only (until other platforms support it)
+		return operatingSystem == "linux"
```

Comment:
Because the `InstallOpts` will conditional change `IsUnprivileged` depending on the operating system that is being tested, the code needs it to be a tristate boolean.

* Unset
* True
* False

In the case that it was never set it will we set to true on Linux and false on Windows (for now). If specifically set then that value is used instead. If we didn't use a tristate then the default of false versus setting false explicitly would be unknown so on Linux it would always be changed to true even on a test where false should be forced. Even if we flip this logic by renaming the variable `Privileged` it results in the same issue.

I also really prefer it to match the command line arguments, all other fields match, don't think we should deviant now.